tests/unit/Database/Validator/CustomIdTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Appwrite\Tests;
+
+use Appwrite\Database\Validator\CustomId;
+use PHPUnit\Framework\TestCase;
+
+class CustomIdTest extends TestCase
+{
+    /**
+     * @var Key
+     */
+    protected $object = null;
+
+    public function setUp(): void
+    {
+        $this->object = new CustomId();
+    }
+
+    public function tearDown(): void
+    {
+    }
+
+    public function testValues()
+    {
+        $this->assertEquals($this->object->isValid('unique()'), true);
+        $this->assertEquals($this->object->isValid('unique)'), false);
+        $this->assertEquals($this->object->isValid('else()'), false);
+        $this->assertEquals($this->object->isValid('dasda asdasd'), false);
+        $this->assertEquals($this->object->isValid('dasda_asdasd'), true);
+        $this->assertEquals($this->object->isValid('asdasdasdas'), true);
+        $this->assertEquals($this->object->isValid('_asdasdasdas'), false);
+        $this->assertEquals($this->object->isValid('as$$5dasdasdas'), false);
+        $this->assertEquals($this->object->isValid(false), false);
+        $this->assertEquals($this->object->isValid(null), false);
+        $this->assertEquals($this->object->isValid('socialAccountForYoutubeSubscribers'), false);
+        $this->assertEquals($this->object->isValid('socialAccountForYoutubeSubscriber'), false);
+        $this->assertEquals($this->object->isValid('socialAccountForYoutubeSubscribe'), true);
+        $this->assertEquals($this->object->isValid('socialAccountForYoutubeSubscrib'), true);
```

Comment:
Maybe a bit more self explanatory strings to test. Seeing all the `asdasd` might be a bit random and doesn't tell me right away what each of those test for 🤔

Reply:
Something like `with space` instead of `dasda asdasd` 😄

Reply:
Agree.